eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Imp/UtxowSpec.hs

```diff
@@ -0,0 +1,168 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Test.Cardano.Ledger.Alonzo.Imp.UtxowSpec (spec) where
+
+import Cardano.Ledger.Alonzo.Core (
+  AlonzoEraTxWits (..),
+  scriptIntegrityHashTxBodyL,
+ )
+import Cardano.Ledger.Alonzo.Plutus.Evaluate (CollectError (..))
+import Cardano.Ledger.Alonzo.Rules (AlonzoUtxosPredFailure (..), AlonzoUtxowPredFailure (..))
+import Cardano.Ledger.Alonzo.Scripts
+import Cardano.Ledger.Alonzo.TxAuxData
+import Cardano.Ledger.Alonzo.TxOut (dataHashTxOutL)
+import Cardano.Ledger.Alonzo.TxWits (Redeemers (..), TxDats (..))
+import Cardano.Ledger.BaseTypes
+import Cardano.Ledger.Core
+import Cardano.Ledger.Plutus
+import Cardano.Ledger.Shelley.LedgerState (esLStateL, lsUTxOStateL, nesEsL, utxosUtxoL)
+import Cardano.Ledger.Shelley.Rules (ShelleyUtxowPredFailure (..))
+import Cardano.Ledger.UTxO (UTxO (..))
+import qualified Data.Map as Map
+import Lens.Micro
+import qualified PlutusLedgerApi.Common as P
+import Test.Cardano.Ledger.Allegra.Imp.UtxowSpec (genInvalidMetadatum)
+import Test.Cardano.Ledger.Allegra.ImpTest (produceScript)
+import Test.Cardano.Ledger.Alonzo.ImpTest (AlonzoEraImp, fixupPPHash)
+import Test.Cardano.Ledger.Imp.Common
+import Test.Cardano.Ledger.Plutus.Examples (guessTheNumber3)
+import Test.Cardano.Ledger.Shelley.ImpTest
+
+spec ::
+  forall era.
+  ( AlonzoEraImp era
+  , TxAuxData era ~ AlonzoTxAuxData era
+  , InjectRuleFailure "LEDGER" ShelleyUtxowPredFailure era
+  , InjectRuleFailure "LEDGER" AlonzoUtxowPredFailure era
+  , InjectRuleFailure "LEDGER" AlonzoUtxosPredFailure era
+  , Arbitrary (Data era)
+  ) =>
+  SpecWith (ImpTestState era)
+spec = describe "UTXOW" $ do
+  it "MissingRedeemers" $ do
```

Comment:
Let's get rid of this one:
* https://github.com/IntersectMBO/cardano-ledger/blob/2c9ba10d085331fa99840a63b40b8e8b716fba5d/libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/AlonzoInvalidTxUTXOW.hs#L175-L184
* https://github.com/IntersectMBO/cardano-ledger/blob/2c9ba10d085331fa99840a63b40b8e8b716fba5d/libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/AlonzoInvalidTxUTXOW.hs#L415-L446